test/Middleware/ImplicitHeadMiddlewareTest.php

```diff
@@ -157,4 +158,47 @@ public function testInvokesHandlerWhenRouteImplicitlySupportsHeadAndSupportsGet(
 
         $this->assertSame($response->reveal(), $result);
     }
+
+    public function testInvokesHandlerWithRequestComposingRouteResultAndAttributes()
+    {
+        $result = $this->prophesize(RouteResult::class);
```

Comment:
From experience it's bad to mock `RouteResult`. It is easy to mock it wrongly and I think it's better to instantiate it using one of `fromRoute` or `fromRouteFailure`. Then the test will be clearer and we can be sure that we have the proper instance.

Reply:
That would mean the other tests are bad as well. This is code I copied from the test before.

Reply:
Yes, not future-proof, and it should be changed imho. I did some of these changes here:
https://github.com/zendframework/zend-expressive-router/commit/adb00caf98d23a930f6378f3652a9a5bc677fc72

As this is a new test it should be fixed before merging, and in separate PR we can fix others.

Don't you think that test will be clearer if you use `fromRoute` or `fromRouteFailure` to create proper `RouteResult` ?

Reply:
It's a complicated test. I don't have time to update all tests. This issue already took me a some time to debug and fix. I'll see what I can do in the weekend.

Reply:
I'm not saying you have to fix all other test, just your test. All others can be fixed later on in separate PR.